File: src/main/java/com/qtech/bubbles/command/TeleportCommand.java
package com.qtech.bubbles.command;

import com.qtech.bubbles.entity.player.PlayerEntity;

public class TeleportCommand implements CommandExecutor {
    @Override
    public boolean execute(PlayerEntity player, String[] args) {
        if (args.length == 2) {
            double x, y;

            try {
                x = Double.parseDouble(args[0]);
                y = Double.parseDouble(args[1]);
            } catch (NumberFormatException exception) {
                try {
                    x = Long.parseLong(args[0]);
                } catch (NumberFormatException exception1) {
                    player.sendMessage("Invalid number for x!");
                    return false;
                }
                try {
                    y = Long.parseLong(args[1]);
                } catch (NumberFormatException exception1) {
                    player.sendMessage("Invalid number for y!");
                    return false;
                }
            }

            player.teleport(x, y);
            return true;
        }
        return false;
    }
}

File: src/main/kotlin/com/qtech/bubbles/command/TeleportCommand.kt
package com.qtech.bubbles.command

import com.qtech.bubbles.common.command.CommandExecutor
import com.qtech.bubbles.entity.player.PlayerEntity

open class TeleportCommand : CommandExecutor {
    override fun execute(player: PlayerEntity, args: Array<String>): Boolean {
        if (args.size == 2) {
            var x: Double
            var y: Double
            try {
                x = args[0].toDouble()
                y = args[1].toDouble()
            } catch (exception: NumberFormatException) {
                x = try {
                    args[0].toLong().toDouble()
                } catch (exception1: NumberFormatException) {
                    player.sendMessage("Invalid number for x!")
                    return false
                }
                y = try {
                    args[1].toLong().toDouble()
                } catch (exception1: NumberFormatException) {
                    player.sendMessage("Invalid number for y!")
                    return false
                }
            }
            player.teleport(x, y)
            return true
        }
        return false
    }
}